components/Inputs/CheckBox.vue

```diff
@@ -2,69 +2,140 @@
  * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
  * See LICENSE for license details.
  */
+
 <template>
-    <Icon
-        :icon="iconClass"
-        size="medium"
-        @click.native="onClick" />
+    <div :class="['checkbox', checkboxStateClasses]">
+        <input
+            :id="associatedLabel"
+            type="checkbox"
+            @input="onValueChange">
+        <label :for="associatedLabel" />
+    </div>
 </template>
 
 <script>
-import nestedVModelMixin from '~/mixins/nestedVModelMixin';
 
 export default {
     name: 'CheckBox',
-    components: {
-        Icon: () => import('~/components/Icon/Icon'),
-    },
-    mixins: [nestedVModelMixin],
     props: {
-        isDisabled: {
+        value: {
+            type: [Boolean, Number],
+            default: false,
+        },
+        disabled: {
             type: Boolean,
-            required: false,
             default: false,
         },
     },
+    data() {
+        return {
+            associatedLabel: '',
+        };
+    },
+    mounted() {
+        // eslint-disable-next-line no-underscore-dangle
+        this.associatedLabel = `checkbox-${this._uid}`;
+    },
     computed: {
-        iconClass() {
-            if (this.isDisabled) {
-                if (this.value) {
-                    return 'sprite-checkbox checkbox-checked--disabled';
-                }
-                return 'sprite-checkbox checkbox-default--disabled';
-            }
-
-            if (Number.isInteger(this.value)) {
-                // Is multi state checkbox
-                switch (this.value) {
-                case 0:
-                    return 'sprite-checkbox checkbox-default';
-                case 1:
-                    return 'sprite-checkbox checkbox-checked';
-                default:
-                    return 'sprite-checkbox checkbox-checked-any';
-                }
-            }
-
-            return this.value
-                ? 'sprite-checkbox checkbox-checked'
-                : 'sprite-checkbox checkbox-default';
+        checkboxStateClasses() {
+            return [
+                {
+                    'checkbox--disabled': this.disabled,
+                    'checkbox--checked': this.value,
+                    'checkbox--checked-any': Number.isInteger(this.value) && this.value === 2,
```

Comment:
if you check the condition by strict equality, it also checks the correctness of the types